src/handlers/psp.ts

```diff
@@ -256,156 +249,113 @@ const initializeProviders = async (state: State): Promise<State> => {
-  });
-  const sponsorWalletsAndTransactionCounts = await Promise.all(sponsorWalletAndTransactionCountPromises);
-  const sponsorWalletsWithSubscriptions = sponsorWalletsAndTransactionCounts.reduce(
-    (acc: SponsorWalletWithSubscriptions[], [logs, data]) => {
-      const sponsorLogOptions = buildLogOptions('additional', { sponsor: data.sponsor }, providerLogOptions);
-
-      utils.logger.logPending(logs, sponsorLogOptions);
-
-      if (isNil(data.sponsorWallet) || isNil(data.transactionCount)) {
-        utils.logger.warn('Failed to fetch sponsor wallet or transaction count', sponsorLogOptions);
-        return acc;
-      }
-
-      return [
-        ...acc,
-        {
-          subscriptions: subscriptionsBySponsor[data.sponsor].map((subscription, idx) => ({
-            ...subscription,
-            nonce: data.transactionCount + idx,
-          })),
-          sponsorWallet: data.sponsorWallet,
-        },
-      ];
-    },
-    []
-  );
-
-  return sponsorWalletsWithSubscriptions;
-};
-
-const submitTransactions = async (state: State) => {
-  const { baseLogOptions, groupedSubscriptions, apiValuesBySubscriptionId, providerStates } = state;
-
-  const providerPromises = providerStates.map(async (providerState) => {
+  const providerSponsorPromises = providerSponsorSubscriptions.map(async (subscriptionGroup) => {
+    const { sponsorAddress, providerState, subscriptions } = subscriptionGroup;
     const { airnodeWallet, providerName, chainId, provider, contracts, voidSigner, currentBlock, gasTarget } =
       providerState;
 
     const providerLogOptions = buildLogOptions('meta', { chainId, providerName }, baseLogOptions);
 
-    // Get subscriptions from template/endnpoint groups
-    const subscriptions = groupedSubscriptions.flatMap((s) => s.subscriptions);
-
-    // Filter subscription by chainId and doblue-check that subscription has an associated API value
-    const chainSubscriptions = subscriptions.filter(
-      (subscription) => subscription.chainId === chainId && apiValuesBySubscriptionId[subscription.id]
+    // Fetch sponsor wallet transaction counts to be able to assign nonces to subscriptions
+    const [transactionCountLogs, walletData] = await getSponsorWalletAndTransactionCount(
+      airnodeWallet,
+      provider,
+      currentBlock,
+      sponsorAddress
     );
 
-    // Check conditions
-    const validSubscriptions = await checkSubscriptionsConditions(
-      chainSubscriptions,
-      apiValuesBySubscriptionId,
-      contracts['DapiServer'],
-      voidSigner,
+    // Skip processing for the current sponsorAddress if the wallet functions fail
+    if (isNil(walletData)) {
+      const sponsorLogOptions = buildLogOptions('additional', { sponsor: sponsorAddress }, providerLogOptions);
+      utils.logger.warn('Failed to fetch sponsor wallet or transaction count', sponsorLogOptions);
+      return;
+    }
+
+    const sponsorWallet = walletData.sponsorWallet;
```

Comment:
It's super minor but this could be `const { sponsorWallet } = walletData;`, feel free to leave it for next time.